core/domain/collection_jobs_one_off.py

```diff
@@ -1,98 +0,0 @@
-# distributed under the License is distributed on an "AS-IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-"""One-off jobs for collections."""
-
-from __future__ import absolute_import  # pylint: disable=import-only-modules
-from __future__ import unicode_literals  # pylint: disable=import-only-modules
-
-import ast
-import logging
-
-from core import jobs
-from core.domain import collection_domain
-from core.domain import collection_services
-from core.platform import models
-import feconf
-import python_utils
-
-(base_models, collection_models,) = models.Registry.import_models([
-    models.NAMES.base_model, models.NAMES.collection])
-
-
-class CollectionMigrationOneOffJob(jobs.BaseMapReduceOneOffJobManager):
-    """A reusable one-time job that may be used to migrate collection schema
-    versions. This job will load all existing collections from the data store
-    and immediately store them back into the data store. The loading process of
-    a collection in collection_services automatically performs schema updating.
-    This job persists that conversion work, keeping collections up-to-date and
-    improving the load time of new collections.
-    """
-
-    _DELETED_KEY = 'collection_deleted'
-    _ERROR_KEY = 'validation_error'
-    _MIGRATED_KEY = 'collection_migrated'
-
-    @classmethod
-    def entity_classes_to_map_over(cls):
-        return [collection_models.CollectionModel]
-
-    @staticmethod
-    def map(item):
-        if item.deleted:
-            yield (CollectionMigrationOneOffJob._DELETED_KEY, 1)
-            return
-
-        # Note: the read will bring the collection up to the newest version.
-        collection = collection_services.get_collection_by_id(item.id)
-        try:
-            collection.validate(strict=False)
-        except Exception as e:
-            logging.exception(
-                'Collection %s failed validation: %s' % (item.id, e))
-            yield (
-                CollectionMigrationOneOffJob._ERROR_KEY,
-                'Collection %s failed validation: %s' % (item.id, e))
-            return
-
-        # Write the new collection into the datastore if it's different from
-        # the old version.
-        #
-        # Note: to_version really should be int, but left as str to conform
-        # with legacy data.
-        if item.schema_version <= feconf.CURRENT_COLLECTION_SCHEMA_VERSION:
-            commit_cmds = [{
-                'cmd': collection_domain.CMD_MIGRATE_SCHEMA_TO_LATEST_VERSION,
-                'from_version': item.schema_version,
-                'to_version': python_utils.UNICODE(
-                    feconf.CURRENT_COLLECTION_SCHEMA_VERSION)
-            }]
-            collection_services.update_collection(
-                feconf.MIGRATION_BOT_USERNAME, item.id, commit_cmds,
```

Comment:
Should we cleanup the update_collection method which handles migration commands?

Reply:
Yes looks like we need to (because some lines are now untested), but I'm not sure how to handle this. Should I just delete all migration-related commands, or delete the function entirely?

Reply:
Maybe change the `update_collection` method such that it doesn't handle migration as an expected command (somewhat around the apply_change_list function (not sure!))